expr/shaper.go

```diff
@@ -155,6 +158,22 @@ func (s *step) castPrimitive(in zcode.Bytes, b *zcode.Builder) *zng.Value {
 	return nil
 }
 
+func (s *step) castUnion(in zcode.Bytes, b *zcode.Builder) *zng.Value {
+	if in == nil {
+		b.AppendNull()
+		return nil
+	}
+	b.BeginContainer()
+	b.AppendPrimitive(zng.EncodeInt(int64(s.toSelector)))
```

Comment:
Feels like there should be some central place (perhaps in package `zng`) where this secret union handshake takes place.

Reply:
I added `zng.BuildUnion` in 982251cbfd6b99acb8e646469308759c3d609780.